prevalence.R

```diff
@@ -83,34 +85,41 @@ first_diag_census <- left_join(
   by = join_by(FirstDiag <= census_date)
 ) |>
   # Exclude records after death
-  filter(is.na(DateOfDeath) | census_date <= floor_date(DateOfDeath, unit = "month"))
+  filter(is.na(DateOfDeath) | census_date <= floor_date(DateOfDeath, unit = "month")) |>
+  filter(is.na(LeftDate) | census_date <= floor_date(LeftDate, unit = "month")) |>
+  filter(is.na(LeftShetlandDate) | census_date <= floor_date(LeftShetlandDate, unit = "month"))
 
 census_data <- first_diag_census |>
   left_join(
     ltc_invite_census,
-    by = join_by(PatientID == PatientID, census_date == census_date),
+    by = join_by(PatientID == PatientID, PracticeID == PracticeID, census_date == census_date),
     multiple = "first"
   ) |>
   left_join(
     ltc_attend_census,
-    by = join_by(PatientID == PatientID, census_date == census_date),
+    by = join_by(PatientID == PatientID, PracticeID == PracticeID, census_date == census_date),
     multiple = "first"
   )
 
 monthly_summary <- census_data |>
-  group_by(census_date) |>
+  group_by(census_date, PracticeID) |>
   summarise(
     count = n_distinct(PatientID),
     ltc_invite_count = sum(!is.na(ltc_invite_date)),
     ltc_attend_count = sum(!is.na(ltc_attend_date)),
-    .groups = "drop"
-  ) |>
-  mutate(census_year = year(census_date)) |>
-  left_join(shetland_pops, by = join_by(closest(census_year >= pop_year))) |>
-  left_join(shetland_list_sizes, by = join_by(closest(census_date >= Date))) |>
-  mutate(
-    pop_prev = count / pop,
-    list_prev = count / list_pop,
     ltc_invite_prop = ltc_invite_count / count,
     ltc_attend_prop = ltc_attend_count / count
-  )
+  ) |>
+  ungroup() |>
+  mutate(census_year = year(census_date))
+
+
+# Can be used once we can identify Practices
+# left_join(shetland_pops, by = join_by(closest(census_year >= pop_year))) |>
+# left_join(shetland_list_sizes, by = join_by(closest(census_date >= Date)))
+# mutate(
+# pop_prev = count / pop,
+# list_prev = count / list_pop,
+# ltc_invite_prop = ltc_invite_count / count,
+# ltc_attend_prop = ltc_attend_count / count
+# )
```

Comment:
I made a small change here as we still need (and can calculate) the attendance and invite percentages.